canonical-web-and-design/maas-ui
# feat(ui): make KVMDetailsHeader component reusable

## Done

- Made KVMDetailsHeader reusable
- Used KVMDetailsHeader to create the individual headers for LXD single, LXD cluster and virsh details pages

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a LXD details page and check that the header looks like the wireframe
- Open a form and check that the header title updates correctly
- Go to a virsh details page and check the same things
- Go to `/MAAS/r/kvm/lxd/cluster/1` and check that the header looks like the wireframe

## Fixes

Fixes canonical-web-and-design/app-squad#363
